reana_client/validation/parameters.py

```diff
@@ -0,0 +1,424 @@
+
+                        # Parse command params
+                        step_commands_params = parse_command_params(step_val)
+                        for command_param in step_commands_params:
+                            cmd_param_steps_mapping[command_param].add(step_name)
+
+            return param_steps_mapping, cmd_param_steps_mapping
+
+        # Yadage parameters
+        param_steps_mapping, cmd_param_steps_mapping = parse_params(
+            self.specification["stages"]
+        )
+        workflow_params = set(param_steps_mapping.keys())
+
+        # REANA input parameters validation
+        self.reana_params_warnings += _warn_not_used_parameters(
+            self.input_parameters, workflow_params
+        )
+
+        # Yadage command and input parameters validation
+        self.workflow_params_warnings = _warn_misused_parameters_in_steps(
+            param_steps_mapping, cmd_param_steps_mapping, "Yadage"
+        )
+
+
+class CWLParameterValidator(ParameterValidatorBase):
+    """REANA CWL workflow parameter validation."""
+
+    def display_messages(self):
+        """Display CWL workflow warnings in console."""
+        self._display_operations_warnings()
+
+    def validate_parameters(self):
+        """Validate input parameters for CWL workflows."""
+
+        def _check_dangerous_operations(workflow):
+            """Check for "baseCommand" and "arguments" in workflow.
+
+            If these keys are found, validate if they have dangerous operations.
+            """
+            cmd_keys = ["baseCommand", "arguments"]
+            for cmd_key in cmd_keys:
+                if cmd_key in workflow:
+                    cmd_values = (
+                        workflow[cmd_key]
+                        if isinstance(workflow[cmd_key], list)
+                        else [workflow[cmd_key]]
+                    )
+                    for cmd_value in cmd_values:
+                        self.operations_warnings += _validate_dangerous_operations(
+                            str(cmd_value), step=workflow.get("id")
+                        )
+
+        cwl_main_spec_path = self.reana_yaml["workflow"].get("file")
+        if os.path.exists(cwl_main_spec_path):
+            run_command(
+                "cwltool --validate --strict {}".format(cwl_main_spec_path),
+                display=False,
+                return_output=True,
+                stderr_output=True,
+            )
+        else:
+            raise ParameterValidationError(
+                "Workflow path {} is not valid.".format(cwl_main_spec_path)
+            )
+
+        workflow = self.specification.get("$graph", self.specification)
+
+        if isinstance(workflow, dict):
+            _check_dangerous_operations(workflow)
+        elif isinstance(workflow, list):
+            for wf in workflow:
+                _check_dangerous_operations(wf)
+
+
+def _warn_not_used_parameters(workflow_parameters, command_parameters, type_="REANA"):
```

Comment:
Since this function is not printing anything anymore perhaps we can change its name to something more meaningful? Maybe something like `get_not_used_parameters_messages` or similar. Same for the similar functions below.

Wouldn't it make sense to move these functions into the base class?